src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/needbraces/InputNeedBracesPatternMatchingForSwitchAllowSingleLine.java

```diff
@@ -84,10 +84,11 @@ case Point(int x, int y ) when (x>=0 && y >=0) :  // violation
                 yield 0;
         };
 
-        int b =  switch (o) {
+        int b;
+        b =  switch (o) {
             case Integer i when (i == 0) -> i;
             case String s when (s.equals("a")) -> s.length();
-            case Point(int x, int y ) when (x>=0 && y >=0) ->  // violation
+            case Point(int x, int y ) when (x>=0 && y >=0) ->
```

Comment:
Please add new tests cases, using all brace options, on input like this:
```java
public class Test {
    
    int m1(int x) {
        return switch (x) {
            case 1 -> 1;
            case 2 -> 2;
            default -> throw new IllegalStateException();
        };
    }

    int m2(int x) {
        return switch (x) {
            case 1 -> { yield 1; }
            case 2 -> { yield 2; }
            default -> { throw new IllegalStateException(); }
        };
    }
}

```

So that we can see the expected behavior for both switch labeled expressions AND blocks in their equivalent forms.

Reply:
done